### PR TITLE
changes after CR3

### PR DESCRIPTION
**Pull request by filippD**

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -8,13 +8,24 @@ def initialize(dim)
...
   	def init_combin(dim)
   		arr = []
-        arr  += Array.new(dim ){|l| Array.new(dim){|i| dim * l + i }}  
-        arr  += Array.new(dim){|l| Array.new(dim){|i|  l + (i*dim)}}
-        arr  += Array.new(1){|l| Array.new(dim){|i|  i+ (i*dim)}}
-        arr  += Array.new(1){|l| Array.new(dim){|i|  i*(dim-1) + (dim-1)}}
+        arr  += Array.new(dim) { |element1| Array.new(dim){|element2| dim * element1 + element2 }}  
+        arr  += Array.new(dim) { |element1| Array.new(dim){|element2|  element1 + (element2*dim)}}
+        arr  += Array.new(1) { |element1| Array.new(dim){|element2|  element2+ (element2*dim)}}
+        arr  += Array.new(1) { |element1| Array.new(dim){|element2|  element2*(dim-1) + (dim-1)}}
```
Comment (heyapricot) [inline lib/board.rb:14]: If you want to make it easier to understand, [Range and step could help with that](https://ruby-doc.org/core-2.6.2/Range.html#method-i-step)

File: lib/game.rb [hunk 2]
```diff
@@ -156,7 +151,7 @@ def move(player)
 
 	def play
 		current_player = @player1
-		until @board.player_won?(@player1.symbol) || @board.player_won?(@player2.symbol) || @board.full? || @wanna_quit
+		until @board.player_won?(current_player.symbol) || @board.full? || @wanna_quit
 			system "clear"
```
Comment (heyapricot) [inline lib/game.rb:155]: This line could be added to the top of `board.print_board`

File: lib/board.rb [hunk 3]
```diff
@@ -58,6 +68,6 @@ def change_matrix(symbol, move_index)
 	end
```
Comment (heyapricot) [inline lib/board.rb:68]: What's the use of `print " "` If you want a line break you can use puts directly. Also, there is string interpolation

Comment (heyapricot) [reply lib/board.rb:68]: [Also, take a look at this stack overflow article about using each_slice .with_index.](https://stackoverflow.com/questions/5983977/ruby-array-each-slice-with-index) Could help removing the `counter`

File: lib/game.rb [hunk 4]
```diff
@@ -138,14 +139,8 @@ def move(player)
...
-				board.change_matrix(player.symbol, move_index)
-			else
-				puts 'Invalid move'
-				move(player)
-			end
+		if board.valid_move?(player_input)
+			board.change_matrix(player.symbol, board.move_index(player_input))
 		elsif player_input.length == 1 && player_input[0].upcase == 'Q'
 			@wanna_quit = true
 		else
```
Comment (heyapricot) [inline lib/game.rb:146]: I don't think it is the responsibility of the `board` object to validate the input. 
Can we split this method? The actions that happen here are:
1. You get input from the user through the console
2. You validate the input
3. If the input is valid you send it to the board, else you request it again
4. You 'write' to the board

**Review by heyapricot — CHANGES_REQUESTED** [review]
# Almost there, let's delegate the right responsibilities to each object

👋🏼 Hello! I have some suggestions about the way the moves are written to the board. Please address my annotations and after that submit a new code review request. Thanks!

File: lib/game.rb (hunk 2 above)
Comment (filippD) [reply lib/game.rb:155]: Which line?

File: lib/board.rb (hunk 3 above)
Comment (filippD) [reply lib/board.rb:68]: " " adds a space before alphabet on the top of the board so it matches the actual positions

**Pull request merged**